### PR TITLE
Support ulysses padding-free grpo

### PR DESCRIPTION
**Pull request by tastelikefeet**
# PR type
- [ ] Bug Fix
- [x] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

Write the detail information belongs to this PR.

## Experiment results

sp=2, completion_length=600

image

image

### CODE REVIEW

File: swift/trainers/arguments.py [hunk 1]
```diff
@@ -221,6 +221,8 @@ class GRPOArgumentsMixin:
     # dataset
     dataset_shuffle: Optional[bool] = True
 
+    padding_free: bool = False
```
Comment (Jintao-Huang) [inline swift/trainers/arguments.py:224]: 移除这里

File: swift/trainers/rlhf_trainer/grpo_trainer.py [hunk 2]
```diff
@@ -1112,6 +1115,73 @@ def _compute_loss(self, model, inputs):
 
         return loss
 
+    @contextmanager
+    def packing_context(self, model: torch.nn.Module):
```
Comment (Jintao-Huang) [inline swift/trainers/rlhf_trainer/grpo_trainer.py:1119]: packing换个名字
